src/System.Net.Sockets/tests/FunctionalTests/TcpClientTest.cs

```diff
@@ -36,16 +39,32 @@ public TcpClientTest(ITestOutputHelper output)
                 string host = System.Net.Test.Common.Configuration.Sockets.SocketServer.IdnHost;
                 int port = System.Net.Test.Common.Configuration.Sockets.SocketServer.Port;
 
-                if (mode == 0)
-                {
-                    await client.ConnectAsync(host, port);
-                }
-                else
+                IPAddress[] addresses;
+                switch (mode)
                 {
-                    IPAddress[] addresses = await Dns.GetHostAddressesAsync(host);
-                    await (mode == 1 ?
-                        client.ConnectAsync(addresses[0], port) :
-                        client.ConnectAsync(addresses, port));
+                    case 0:
+                        await client.ConnectAsync(host, port);
+                        break;
+                    case 1:
+                        addresses = await Dns.GetHostAddressesAsync(host);
+                        await client.ConnectAsync(addresses[0], port);
+                        break;
+                    case 2:
+                        addresses = await Dns.GetHostAddressesAsync(host);
+                        await client.ConnectAsync(addresses, port);
+                        break;
+
+                    case 3:
+                        await Task.Factory.FromAsync(client.BeginConnect, client.EndConnect, host, port, null);
+                        break;
+                    case 4:
+                        addresses = await Dns.GetHostAddressesAsync(host);
+                        await Task.Factory.FromAsync(client.BeginConnect, client.EndConnect, addresses[0], port, null);
+                        break;
+                    case 5:
+                        addresses = await Dns.GetHostAddressesAsync(host);
+                        await Task.Factory.FromAsync(client.BeginConnect, client.EndConnect, addresses, port, null);
+                        break;
```

Comment:
> This test is getting a little silly. :) Can we break it out into separate test methods?

+1  Definitely a [code smell](https://en.wikipedia.org/wiki/Code_smell).

There are two test principles broken:
1. Test is not written as a functional spec (name should be enough to derive the intent)
2. The intent of the test isn't clear and the code is unnecessarily complicated.